### PR TITLE
Ft oop code conversion

### PR DESCRIPTION
**Pull request by Code-250**
Hello I have managed to convert my api into OOP codes 
by that 

- I am able to perform CRUD operations to posts
- I have managed to protect routes
- managed to log and register users
- able to delete user.

### CODE REVIEW

File: Src/App/Helpers/validation.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import Joi from "joi";
+
+const createPost = Joi.object().keys({
```
Comment (RedJanvier) [inline Src/App/Helpers/validation.js:3]: Can you also convert this file to a class?

File: Src/App/Middlewares/authMidleware.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import jwt from "jsonwebtoken";
+import errorHandler from "../utils/errorHandler.js";
+
+export const routeProtect = (req, res, next) => {
```
Comment (RedJanvier) [inline Src/App/Middlewares/authMidleware.js]: This must also be a class not a standalone function.

File: Src/App/utils/errorHandler.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+export default (res, status, messsage) => {
```
Comment (RedJanvier) [inline Src/App/utils/errorHandler.js]: Can you combine this and successHandler into a single function please?

File: Src/App/Controllers/authController.js [hunk 4]
```diff
@@ -1,34 +1,118 @@
...
+          res,
+          400,
+          `Oh No! Error while logging user :( ${err.message}`
+        )
+      );
+    }
+  }
+  // get all the users
+
+  static async getUsers(req, res, next) {
```
Comment (RedJanvier) [inline Src/App/Controllers/authController.js:91]: Why are you not catching errors in this controller?

File: Src/App/Controllers/authController.js [hunk 5]
```diff
@@ -1,34 +1,118 @@
...
+    if (!user) {
+      return next(
+        errorHandler(res, 404, "Ooops! Looks like there are no users :(")
+      );
+    }
+    return successHandler(res, 200, "All users retreived successfully", user);
+  }
+  // remove Users
+
+  static async deleteUser(req, res, next) {
```
Comment (RedJanvier) [inline Src/App/Controllers/authController.js:113]: You need to catch errors that might occur!

File: .env [hunk 6]
```diff
@@ -1,2 +1,3 @@
 PORT = 8080
```
Comment (RedJanvier) [inline .env]: Please remove this file! This is a necessity now on. Your PR will not be merged if this file is in it!

**Review by RedJanvier — CHANGES_REQUESTED** [review]
You are not catching errors in some of your controllers please fix it ASAP!

**Review by RedJanvier — APPROVED** [review]

**Review by Procech20 — APPROVED** [review]
Looks good to me!

**Pull request merged**